Добавлены тесты в ветку develop

DamsYaYa opened the pull request.

DamsYaYa committed:
Добавлены тесты в ветку develop

nikolay-977 commented on src/test/java/example/CatTest.java:
```diff
@@ -0,0 +1,39 @@
+        Assert.assertEquals(expectedSound, actualSound);
+    }
+
+    @Mock Predator predator;
```
Необходимо исправить: В начале класса следует описывать переменные, а затем конструкторы и методы. Также можно установить курсор на корневую директорию и нажать сочетание клавиш Ctrl+Alt+L (Option+Command+L для Mac), затем выбрать checkbox rearrange entries и нажать ok.

nikolay-977 commented on src/test/java/example/ParametrizedTestLion.java:
```diff
@@ -0,0 +1,40 @@
+///parametrized Lion test
+
+@RunWith(Parameterized.class)
+
```
Необходимо исправить: не следует использовать пропуски строк после аннотаций.

nikolay-977 commented on src/main/java/example/Lion.java:
```diff
@@ -0,0 +1,34 @@
+
+public class Lion {
+
+    boolean hasMane;
```
Можно улучшить: Для всех полей следует указывать модификаторы доступа.

nikolay-977 commented on src/main/java/example/Lion.java:
```diff
@@ -0,0 +1,34 @@
+            throw new Exception("Используйте допустимые значения пола животного - самей или самка");
+        }
+    }
+
```
Можно улучшить: лишних пропусков строк следует избегать. Для зрительного разделения кода достаточно использовать один пропуск строки. Пропуски строк должны быть однообразными.

nikolay-977 commented on src/test/java/example/LionTest.java:
```diff
@@ -0,0 +1,56 @@
+        } catch (Exception ex) {
+            exception = ex;
+        }
+        Assert.assertNotNull(exception);
```
Можно улучшить:  проверку лучше выполнять внутри блока catch конструкции try-catch

```
try {
    // ...
} catch (Exception e) {
    // assertions
}
```

Либо использовать специальные методы. Примеры проверок exception:
https://www.baeldung.com/junit-assert-exception
https://www.baeldung.com/assertj-exception-assertion

nikolay-977 reviewed (commented):
Можно улучшить: хорошей практикой является добавление README.md с описанием проекта, которое включает название, технологии в проекте (например, Java 11, JUnit 4.13.2, maven 3.9.0 и т.д.), настройка (если необходима) и как запускать (`mvn clean test` - кавычки позволяют выполнять запуск из файла README.md). Как оформлять README.md можно ознакомиться в статье https://blog.jetbrains.com/idea/2022/08/markdown-support-improvements/

DamsYaYa committed:
Добавлен файл README.md
Удалены лишние пробелы и переносы строк
Перенесены описания переменных
Добавлен модификатор доступа для полей
